plugins/inputs/sql/sql.go

```diff
@@ -462,12 +462,10 @@ func (q *query) parse(acc telegraf.Accumulator, rows *dbsql.Rows, t time.Time, l
 			if q.fieldFilterInt.Match(name) {
 				v, err := internal.ToInt64(columnData[i])
 				if err != nil {
-					if err != nil {
-						if !errors.Is(err, internal.ErrOutOfRange) {
-							return 0, fmt.Errorf("converting field column %q to int failed: %w", name, err)
-						}
-						logger.Warnf("field column %q: %v", name, err)
+					if !errors.Is(err, internal.ErrOutOfRange) {
+						return 0, fmt.Errorf("converting field column %q to int failed: %w", name, err)
 					}
+					logger.Warnf("field column %q: %v", name, err)
 				}
```

Comment:
This is just a personal preference, but avoiding nested if statements can improve readability and maintainability by reducing indentation. Feel free to ignore if it doesn't fit your style!

```suggestion
				if err != nil && !errors.Is(err, internal.ErrOutOfRange) {
    return 0, fmt.Errorf("converting field column %q to int failed: %w", name, err)
}

if err != nil {
    logger.Warnf("field column %q: %v", name, err)
}

```

Reply:
The current structure makes it easier to "ignore" the error case when reading the code and makes sure that the error is handled in any case. With your change, the reader needs to follow both if conditions and then needs to figure out that both handle error cases. Long story short, I prefer the current pattern over your suggestion.